docs/examples/vectorfields.py

```diff
@@ -0,0 +1,493 @@
+# recreating the results by [Berlinghieri et. al, (2023)](https://arxiv.org/pdf/2302.10364.pdf).
+# %%
+from jax.config import config
+
+config.update("jax_enable_x64", True)
+import torch
+
+torch.manual_seed(123)
+import jax.random as jr
+import jax.numpy as jnp
+from jaxtyping import install_import_hook
+import matplotlib.pyplot as plt
+from matplotlib import rcParams
+from mpl_toolkits.axes_grid1 import make_axes_locatable
+import numpy as onp
+import optax as ox
+import tensorflow_probability as tfp
+from jax.config import config
+from dataclasses import dataclass
+from jaxtyping import (
+    Array,
+    Float,
+    install_import_hook,
+)
+from jax import hessian, jit
+
+with install_import_hook("gpjax", "beartype.beartype"):
+    import gpjax as gpx
+
+# Enable Float64 for more stable matrix inversions.
+key = jr.PRNGKey(123)
+plt.style.use(
+    "https://raw.githubusercontent.com/JaxGaussianProcesses/GPJax/main/docs/examples/gpjax.mplstyle"
+)
+cols = rcParams["axes.prop_cycle"].by_key()["color"]
+
+# jnp.set_printoptions(4, edgeitems=30, linewidth=100000,
+#     formatter=dict(float=lambda x: "%.3g" % x))
+
+
+# %% [markdown]
+# ## Data Loading
+# Data and original code are publicly available at https://github.com/renatoberlinghieri/Helmholtz-GP
+#
+# The data are 297 (artificial) measurements of ocean velocity, given by the vector field
+#
+# $$
+# \mathbf{F}(\mathbf{x}) = -x^{(1)}\hat{\imath} + x^{(0)}\hat{\jmath},
+# $$
+#
+#
+#
+# where $\mathbf{x} = (x^{(0)}$,$x^{(1)})^\text{T}$, $\hat{\imath}$ and $\hat{\jmath}$ are unit vectors in the standard Cartesian directions (dimensions will be indicated by superscripts).
+#
+# The measurements are contained in a dataset $D_0=\left\{ \left(\mathbf{x}_{0,i} , \mathbf{y}_{0,i} \right)\right\}_{i=1}^N$, where $\mathbf{y}_i$ is a 2 dimensional velocity vector and $\mathbf{x}_i$ is a 2 dimensional position vector on an $N=17\times17$ grid, equally spaced over the interval $[-1,1] \times[-1,1]$. 8 measurements of $\mathbf{F}(\mathbf{x})$ at new locations are contained in a dataset $D_T=\left\{\left(\mathbf{x}_{T,i}, \mathbf{y}_{T,i} \right)\right\}_{i=1}^8$, which are allocated to training data.
+#
+#
+
+# %%
+# loading in data
+try:
+    Pos_Train = onp.genfromtxt("data/XY_train_vortex.csv", delimiter=",").T
+    Pos_Test = onp.genfromtxt("data/XY_test_vortex.csv", delimiter=",").T
+    Vel_Train = onp.genfromtxt("data/UV_train_vortex.csv", delimiter=",").T
+    Vel_Test = onp.genfromtxt("data/UV_test_vortex.csv", delimiter=",").T
+except FileNotFoundError:
+    Pos_Train = onp.genfromtxt(
+        "docs/examples/data/XY_train_vortex.csv", delimiter=","
+    ).T
+    Pos_Test = onp.genfromtxt("docs/examples/data/XY_test_vortex.csv", delimiter=",").T
+    Vel_Train = onp.genfromtxt(
+        "docs/examples/data/UV_train_vortex.csv", delimiter=","
+    ).T
+    Vel_Test = onp.genfromtxt("docs/examples/data/UV_test_vortex.csv", delimiter=",").T
```

Comment:
How about something like the following to make it more concise and easier to follow?
```python
from collections import namedtuple

PosVel = namedtuple("PosVel", ["pos", "vel"])

def load_train_test(path):
    pos_train = onp.genfromtxt(f"{path}/XY_train_vortex.csv", delimiter=",").T
    vel_train = onp.genfromtxt(f"{path}/UV_train_vortex.csv", delimiter=",").T
    data_train = PosVel(pos_train, vel_train)
    pos_test = onp.genfromtxt(f"{path}/XY_test_vortex.csv", delimiter=",").T
    vel_test = onp.genfromtxt(f"{path}/UV_test_vortex.csv", delimiter=",").T
    data_test = PosVel(pos_test, vel_test)
    return data_train, data_test

try:
    data_train, data_test = load_train_test("data")  # notebook run from within docs/examples/
except FileNotFoundError:
    data_train, data_test = load_train_test("docs/examples/data")  # notebook run from repository base directory
```

then below you could simplify the plotting as follows:
```python
def plot_vector_field(ax, pos_vel, **kwargs):
    ax.quiver(pos_vel.pos[0], pos_vel.pos[1], pos_vel.vel[0], pos_vel.vel[1], **kwargs)

fig, ax = plt.subplots(1, 1)
plot_vector_field(ax, data_test, color=colors[0], label="$D_0$")
plot_vector_field(ax, data_train, color=colors[1], alpha=0.7, label="$D_T$")
```

Reply:
Could even rewrite the `load_train_test` as follows:
```python
def load_vector_field(pos_file, vec_file):
    pos = onp.genfromtxt(pos_file, delimiter=",").T
    vel = onp.genfromtxt(vel_file, delimiter=",").T
    return PosVel(pos, vel)

def load_train_test(path):
    data_train = load_vector_field(f"{path}/XY_train_vortex.csv", f"{path}/UV_train_vortex.csv")
    data_test = load_vector_field(f"{path}/XY_test_vortex.csv", f"{path}/UV_test_vortex.csv")
    return data_train, data_test
```
but maybe that's overkill (:

Reply:
This isn't implemented anymore, using a real dataset that is in a single csv.